libs/Common/Types.inl

```diff
@@ -2147,7 +2147,11 @@ void TImage<TYPE>::toGray(TImage<T>& out, int code, bool bNormalize, bool bSRGB)
 		out.create(rows, cols);
 	ASSERT(cv::Mat::isContinuous());
 	ASSERT(out.cv::Mat::isContinuous());
+#ifdef __clang__
 	const int scn(cv::Mat::channels());
+#else
```

Comment:
GCC 7.4.3 was throwing an error on this line.

```
libs/MVS/../Common/Types.inl:2150:33: error: cannot call member function 'int cv::Mat::channels() const' without object
```
I suppose this could be a GCC version problem, OpenCV version problem, or something else entirely. I'm just putting this in here now as something that solves the problem in our case, and solicit your advice. We can refine this `#ifdef` to be something closer to the actual trigger or remove this from the PR as out of scope if you would like.

Reply:
try using `this->` like `this->cv::Mat::channels()`

Reply:
I changed to this and it worked on all of our platforms, thanks.